lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -98,6 +103,9 @@ def __init__(self, charm, peer_relation_name: str):
         self.config = YamlConfigSetter(base_path=self.paths.conf)
         self._charm = charm
         self._peer_relation_name = peer_relation_name
+        self.perf_profile = OpenSearchPerfProfile.from_str(
```

Comment:
I don't think we need this - we can call `apply_perf_templates_if_needed` directly with the new profile